File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.py
# 최대 공약수 / 유클리드 호재법
def gcd(n, m):
    if m == 0:
        return n
    return gcd(m, n % m) if n > m else gcd(n, m % n);

    
    
def solution(numer1, denom1, numer2, denom2):
    answer = [denom1*numer2+denom2*numer1, denom1*denom2]
    g = gcd(answer[0], answer[1])
    return [answer[0]//g, answer[1]//g]
    

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int bj = numer1 * denom2 + numer2 * denom1;
        int bm = denom1 * denom2;

        return new int[]{bj / gcd(bj, bm), bm / gcd(bj, bm)};
    }

    public int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return a > b ? gcd(b, a % b) : gcd(a, b % a);
    }
}